src/blockdata/transaction.rs

```diff
@@ -262,6 +255,58 @@ impl Default for TxOut {
     }
 }
 
+/// Result of [`Transaction::encode_signing_data_to`].
+///
+/// This type forces the caller to handle SIGHASH_SINGLE bug corner case.
+pub enum EncodeSigningDataResult<E> {
+    /// Input data is an instance of `SIGHASH_SINGLE` bug
+    SighashSingleBug,
+    /// Operation performed normally.
+    Result(Result<(), E>),
+}
+
+impl<E> EncodeSigningDataResult<E> {
+    /// Caller is recommended to use the following pattern to handle this corner case:
```

Comment:
A summary would be nice + maybe reword/expand this. Something like:

> Checks for `SIGHASH_SINGLE` bug returning error if the writer failed.
>
> This is method is provided for easy and correct handling of the result because `SIGHASH_SINGLE` bug is a special case that must not be ignored nor cause panicking. Since the data is usually written directly into a hasher which never fails, the recommended pattern to handle this is: